### PR TITLE
Change request

### PR DESCRIPTION
**Pull request by GranadaBrayan**
The changes suggested by the tutor for the views are made, as well as the implementation of the Devise gem functionalities.

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+GMAIL_USERNAME = blogpassreset@gmail.com
```
Comment (estebancordoba) [inline .env]: This is sensitive data, it shouldn't be in the repo. You could create an `env.sample` file to give an example of what the `.env` should look like (also include instructions in the README)

File: app/helpers/articles_helper.rb [hunk 2]
```diff
@@ -1,2 +1,21 @@
 module ArticlesHelper
+  def show_article?(article)
```
Comment (estebancordoba) [inline app/helpers/articles_helper.rb:2]: great

File: app/helpers/articles_helper.rb [hunk 3]
```diff
@@ -1,2 +1,21 @@
...
+    current_user.is_owner?(article) || article.status == 'public'
```
Comment (estebancordoba) [inline app/helpers/articles_helper.rb]: the string 'public' could be a constant

File: app/mailers/application_mailer.rb [hunk 4]
```diff
@@ -1,4 +1,4 @@
 class ApplicationMailer < ActionMailer::Base
-  default from: 'from@example.com'
+  default from: 'blogpassreset@gmail.com'
```
Comment (estebancordoba) [inline app/mailers/application_mailer.rb]: It isn't a good idea to include emails directly in the code, it's better to bring the data from the `.env`

File: app/views/articles/index.html.erb [hunk 5]
```diff
@@ -7,7 +7,7 @@
     <div class="col-lg-3"></div>
     <div class="col-lg-6">
       <% @articles.each do |article| %>
-        <%if article.user == current_user || article.status == 'public'%>
+        <%if show_article?(article)%>
```
Comment (estebancordoba) [inline app/views/articles/index.html.erb]: `<% if show_article?(article) %>`

File: app/views/articles/index.html.erb [hunk 6]
```diff
@@ -31,8 +31,8 @@
                   <div class="col-4">
                   </div>
                   <div class="col-4">
-                    <%if current_user != article.user %>
-                      <%if current_user.followings.include?(article.user) %>
+                    <%unless owner_article(article) %>
```
Comment (estebancordoba) [inline app/views/articles/index.html.erb]: don't forget the blanks

File: config/initializers/devise.rb [hunk 7]
```diff
@@ -24,10 +24,10 @@
   # Configure the e-mail address which will be shown in Devise::Mailer,
   # note that it will be overwritten if you use your own mailer class
   # with default "from" parameter.
-  config.mailer_sender = 'please-change-me-at-config-initializers-devise@example.com'
+  config.mailer_sender = 'blogpassreset@gmail.com'
```
Comment (estebancordoba) [inline config/initializers/devise.rb]: sensitive data

File: package.json [hunk 8]
```diff
@@ -6,7 +6,7 @@
     "@rails/activestorage": "^6.0.0",
     "@rails/ujs": "^6.0.0",
     "@rails/webpacker": "5.4.0",
-    "bootstrap": "^5.1.0",
+    "bootstrap": "4.0.0-alpha.6",
```
Comment (estebancordoba) [inline package.json:9]: why the version change?

**Review by estebancordoba — CHANGES_REQUESTED** [review]

File: .env (hunk 1 above)
Comment (GranadaBrayan) [reply .env]: I understand, I will make the change to not compromise sensitive data, I will add the .env file to the .gitignore file.

File: app/helpers/articles_helper.rb (hunk 2 above)
Comment (GranadaBrayan) [reply app/helpers/articles_helper.rb:2]: Thanks

File: app/helpers/articles_helper.rb (hunk 3 above)
Comment (GranadaBrayan) [reply app/helpers/articles_helper.rb]: Excellent idea, thanks.

File: app/mailers/application_mailer.rb (hunk 4 above)
Comment (GranadaBrayan) [reply app/mailers/application_mailer.rb]: I'll make the change.

File: app/views/articles/index.html.erb (hunk 5 above)
Comment (GranadaBrayan) [reply app/views/articles/index.html.erb]: I'll make the change.

File: app/views/articles/index.html.erb (hunk 6 above)
Comment (GranadaBrayan) [reply app/views/articles/index.html.erb]: I'll make the change.

File: config/initializers/devise.rb (hunk 7 above)
Comment (GranadaBrayan) [reply config/initializers/devise.rb]: I'll make the change.

File: package.json (hunk 8 above)
Comment (GranadaBrayan) [reply package.json:9]: By mistake I made the change before, initially, I work with bootstrap 4.

File: .env (hunk 1 above)
Comment (estebancordoba) [reply .env]: remove the `.env` file from the repo

**Review by estebancordoba — CHANGES_REQUESTED** [review]

**Review by estebancordoba — APPROVED** [review]

**Pull request merged**